### DISCUSSION TITLE
How does Chezmoi pick a shell for its cd command?

### DISCUSSION THREAD

**Discussion by J-Kappes** [twpayne/chezmoi#3088] [Q&A] [answered]
I recently switched to nushell and my `.profile` file specifies it but when I `cm cd` from within nushell in both Konsole and Alacritty I wind up in Zsh.

**Comment by twpayne** [comment, marked as answer]
See https://github.com/twpayne/chezmoi/blob/master/pkg/shell/shell_posix.go#L14.

chezmoi checks the following in order, using the first value that it finds:
1. The `$SHELL` environment variable.
2. The `getpwnam_r` library call.
3. The output of `getent passwd $USER`.
4. The user's entry in `/etc/passwd`.
5. The default shell for the OS.

Are you sure you changed your shell with `chsh`?

**Reply by J-Kappes** [reply]
> Are you sure you changed your shell with `chsh`?

I did not. I hadn't heard of that tool before. But should I use it? I use a locally compiled nushell from my home directory and `chsh`  will only let me pick shells that are listed in `/etc/shells`. Is it OK to list list my nushell there?

Anyway the reason seems to be that `$SHELL` variable is still set to Zsh.

**Reply by twpayne** [reply]
nushell has [instructions for setting it as the default shell](https://www.nushell.sh/book/default_shell.html#setting-nu-as-login-shell-linux-bsd-macos). You should follow them. This is not related to chezmoi.
